tokio/src/time/driver/sleep.rs

```diff
@@ -244,40 +297,120 @@ impl Sleep {
     ///
     /// [`Pin::as_mut`]: fn@std::pin::Pin::as_mut
     pub fn reset(self: Pin<&mut Self>, deadline: Instant) {
-        let me = self.project();
-        me.entry.reset(deadline);
-        *me.deadline = deadline;
+        self.reset_inner(deadline)
+    }
+
+    cfg_trace! {
+        fn reset_inner(self: Pin<&mut Self>, deadline: Instant) {
+            let me = self.project();
+
+            let now = me.inner.time_source.now();
+            let duration = me.inner.time_source.deadline_to_tick(deadline) - now;
+            let _span_guard = me.inner.resource_span.enter();
+            tracing::trace!(
+                target: "tokio::resource::state_update",
+                duration = duration,
+                duration.unit = "ms",
+                duration.op = "override",
+            );
+            me.entry.reset(deadline);
+            (*me.inner).deadline = deadline;
+        }
     }
 
-    fn poll_elapsed(self: Pin<&mut Self>, cx: &mut task::Context<'_>) -> Poll<Result<(), Error>> {
-        let me = self.project();
+    cfg_not_trace! {
+        fn reset_inner(self: Pin<&mut Self>, deadline: Instant) {
+            let me = self.project();
+            me.entry.reset(deadline);
+            (*me.inner).deadline = deadline;
+        }
+    }
```

Comment:
Seems there is a problem with the `trace!` macro when setting both the parent and the target at the same time: 


It results in: 

```
error: expected type, found `"runtime::resource::state_update"`
   --> tokio/src/time/driver/sleep.rs:319:25
    |
319 |                 target: "runtime::resource::state_update",
    |                       - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected type
    |                       |
    |                       tried to parse a type due to this type ascription
    |
    = note: `#![feature(type_ascription)]` lets you annotate an expression with a type: `<expr>: <type>`
    = note: see issue #23416 <https://github.com/rust-lang/rust/issues/23416> for more information

```

Reply:
oh, i think i got the order backwards, it should be `target: "...", parent: ...`, rather than `parent: ..., target: "..."`. because the order is wrong, it doesn't match the expected macro match arm, and the compiler tries to parse `target` as a variable name with a type ascription. 

swapping the order should fix this, my bad!